### ISSUE TITLE
Arch::name() unused.

### ISSUE THREAD

**Issue by vext01** [CensoredUsername/dynasm-rs#93]
I noticed:

```
warning: method `name` is never used
  --> plugin/src/arch/mod.rs:13:8
   |
12 | pub(crate) trait Arch : Debug + Send {
   |                  ---- method in this trait
13 |     fn name(&self) -> &str;
   |        ^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: field `name` is never read
  --> plugin/src/arch/mod.rs:22:5
   |
21 | pub struct DummyArch {
   |            --------- field in this struct
22 |     name: &'static str
   |     ^^^^
   |
   = note: `DummyArch` has derived impls for the traits `Debug` and `Clone`, but these are intentionally ignored during dead code analysis
```

Did you want to kill the method entirely?

**Comment by CensoredUsername** [comment]
It's plugin internal, so you can just kill it.

**Comment by vext01** [comment]
Indeed. I just wanted to check I wasn't bulldozing something you wanted to keep around. Thanks.

**Comment by CensoredUsername** [comment]
Fixed in v3.0.0.
